gph/src/ripser.h

```diff
@@ -691,9 +691,32 @@ class ripser
         };
 
         const index_t cnt = n - (k - 1);
+
         if (pred(n) || (cnt <= 0))
             return n;
-        return get_max(n, cnt, pred);
+
+        if (k != 3) {
+            return get_max(n, cnt, pred);
+        } else {
+            long double to_cbrt = 6 * idx;
+            index_t guess = static_cast<index_t>(std::floor(std::cbrt(to_cbrt)));
+            if (!pred(guess)) {
+                while (true) {
+                    --guess;
+                    if (pred(guess))
+                        break;
+                }
+            } else {
+                while (true) {
+                    ++guess;
+                    if (!pred(guess)) {
+                        --guess;
+                        break;
+                    }
+                }
+            }
+            return guess;
```

Comment:
Suggestion:

```cpp
bool increment = pred(guess);

while (true) {
    if (increment) {
	++guess;
	if (!pred(guess)) {
	    --guess;
	    break;
	}
    } else {
	--guess;
        if (pred(guess))
            break;
    }
}

return guess;
```

Or 

```cpp
bool increment = pred(guess);

while (true) {
    if (increment) {
	++guess;
	if (!pred(guess)) {
	    break;
	}
    } else {
	--guess;
        if (pred(guess))
            break;
    }
}

return increment ? guess - 1 : guess;
```

Reply:
Don't these solutions hard-code the assumptions that the right answer is never 1 step away from the initial guess?

Reply:
No, we have first the `while(true)` loop and then we compute the `if` condition. We can increase/decrease multiple times until one of the break condition is true.

Reply:
Ah, sorry, I mis-read it.  The first solution is certainly right.  What's going on in the second one and why should we prefer it?

Reply:
It is not better IMO. It just keeps the core of the loop almost the same. And the `--guess` performed just before the break in the `increment` case is done at the return step.

It just keeps the "same" logic inside the loop, nothing else. A matter of taste, and also both solutions are possible.

Reply:
Ok!  I'll go with the first one for now just because it seems more human-friendly to me.

Reply:
Done!